### ISSUE TITLE
google-protobuf: installation fails due to switch (VALUE)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1917]
```
truffleruby 20.1.0-dev-daa9b865, like ruby 2.6.5, GraalVM CE Native [x86_64-darwin]
```

Error:

```
$ gem install grpc
Fetching google-protobuf-3.11.4.gem
Fetching googleapis-common-protos-types-1.0.4.gem
Fetching grpc-1.27.0.gem
Building native extensions. This could take a while...
ERROR:  Error installing grpc:
	ERROR: Failed to build gem native extension.

    current directory: ~/.gem/truffleruby/2.6.5/gems/google-protobuf-3.11.4/ext/google/protobuf_c
~/.rubies/truffleruby-dev/bin/truffleruby -I ~/.rubies/truffleruby-dev/lib/mri -r ./siteconf20200218-63349-ay6j4q.rb extconf.rb
creating Makefile

current directory: ~/.gem/truffleruby/2.6.5/gems/google-protobuf-3.11.4/ext/google/protobuf_c
make "DESTDIR=" clean

current directory: ~/.gem/truffleruby/2.6.5/gems/google-protobuf-3.11.4/ext/google/protobuf_c
make "DESTDIR="
compiling protobuf.c
In file included from protobuf.c:31:
In file included from ./protobuf.h:34:
In file included from ~/.rubies/truffleruby-dev/lib/cext/include/ruby/ruby.h:2446:
~/.rubies/truffleruby-dev/lib/cext/include/truffleruby/truffleruby.h:258:13: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  const int n_mand = pre + post;
            ^
1 warning generated.
compiling defs.c
In file included from defs.c:33:
In file included from ./protobuf.h:34:
In file included from ~/.rubies/truffleruby-dev/lib/cext/include/ruby/ruby.h:2446:
~/.rubies/truffleruby-dev/lib/cext/include/truffleruby/truffleruby.h:258:13: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  const int n_mand = pre + post;
            ^
1 warning generated.
compiling storage.c
In file included from storage.c:31:
In file included from ./protobuf.h:34:
In file included from ~/.rubies/truffleruby-dev/lib/cext/include/ruby/ruby.h:2446:
~/.rubies/truffleruby-dev/lib/cext/include/truffleruby/truffleruby.h:258:13: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  const int n_mand = pre + post;
            ^
1 warning generated.
compiling message.c
In file included from message.c:31:
In file included from ./protobuf.h:34:
In file included from ~/.rubies/truffleruby-dev/lib/cext/include/ruby/ruby.h:2446:
~/.rubies/truffleruby-dev/lib/cext/include/truffleruby/truffleruby.h:258:13: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  const int n_mand = pre + post;
            ^
1 warning generated.
compiling repeated_field.c
In file included from repeated_field.c:31:
In file included from ./protobuf.h:34:
In file included from ~/.rubies/truffleruby-dev/lib/cext/include/ruby/ruby.h:2446:
~/.rubies/truffleruby-dev/lib/cext/include/truffleruby/truffleruby.h:258:13: warning: ISO C90 forbids mixing declarations and code [-Wdeclaration-after-statement]
  const int n_mand = pre + post;
            ^
repeated_field.c:131:7: error: statement requires expression of integer type ('VALUE' (aka 'void *') invalid)
      switch (rb_range_beg_len(arg, &beg, &len, self->size, 0)) {
      ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
make: *** [repeated_field.o] Error 1

make failed, exit code 2

Gem files will remain installed in ~/.gem/truffleruby/2.6.5/gems/google-protobuf-3.11.4 for inspection.
Results logged to ~/.gem/truffleruby/2.6.5/extensions/x86_64-darwin/20.1.0-dev-daa9b865/google-protobuf-3.11.4/gem_make.out
```

**Comment by eregon** [comment]
Right, that's the same error as in https://github.com/oracle/truffleruby/issues/1837#issuecomment-560100365 but since it [wasn't specific to protobuf](https://github.com/oracle/truffleruby/issues/1837#issuecomment-581021059) I closed that issue.
This can track making `switch (VALUE)` work then.

**Comment by eregon** [comment]
I fixed `switch (VALUE)` to compile in 0552ac72e3396492dc42ea53c1ac5ee2eb24a3ba.
`gem install google-protobuf` works now.

The grpc `extconf.rb` fails though with:
```
extconf.rb:28:in `<main>': undefined method `+' for nil:NilClass (NoMethodError)
```
code:
```ruby
if ENV['AR'].nil? || ENV['AR'].size == 0
    ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
end
```
Looks like we need a value for `RbConfig::CONFIG['AR']`.
I'll split that to a different issue.
